pkg/services/sqlstore/preferences.go

```diff
@@ -120,3 +134,53 @@ func (ss *SQLStore) SavePreferences(ctx context.Context, cmd *models.SavePrefere
 		return err
 	})
 }
+
+func (ss *SQLStore) UpsertPreferences(ctx context.Context, cmd *models.PatchPreferencesCommand) error {
```

Comment:
Nit. I understand why you name it this way, but you already have `SavePreferences` doing an upsert so can be confusing for the consumer. This one does an insert or a patch so might make more sense to call it `PatchPreferences`. An alternative would be to add a comment like this:

```suggestion
// UpsertPreferences inserts or patches provided non-nil preferences.
func (ss *SQLStore) UpsertPreferences(ctx context.Context, cmd *models.PatchPreferencesCommand) error {
```